integration/spark/shared/src/main/java/io/openlineage/spark/api/SparkOpenLineageConfig.java

```diff
@@ -32,15 +34,18 @@ public class SparkOpenLineageConfig extends OpenLineageConfig<SparkOpenLineageCo
 
   public static final String DEFAULT_NAMESPACE = "default";
 
-  @Setter @NonNull private String namespace;
-  @Setter @Getter private String parentJobName;
-  @Setter @Getter private String parentJobNamespace;
-  @Setter @Getter private String parentRunId;
-  @Setter @Getter private String overriddenAppName;
-  @Setter @NonNull private String debugFacet;
-  @Setter @Getter private String testExtensionProvider;
-  @Setter private JobNameConfig jobName;
-  @Setter private JobConfig job;
+  @NonNull private String namespace;
+  private String parentJobName;
+  private String parentJobNamespace;
+  private String parentRunId;
+  private String overriddenAppName;
+  @NonNull private String debugFacet;
+  private String testExtensionProvider;
+  private JobNameConfig jobName;
+  private JobConfig job;
+
+  @JsonProperty("columnLineage")
+  private ColumnLineageConfig columnLineageConfig;
```

Comment:
Would it make sense to have it as a part of `DatasetConfig`?

Reply:
It is often said that nothing is more permanent than s temporary change, but I hope it won't be this case. Moving it to `DatasetConfig` would require extending this class for Spark integration (because we don't have this facet in Java) and it carries the risk introducing some regression. Additionally, it would take extra 30-60 minutes of extra work that could be better spent elsewhere. Is it worth it?